src/commands/deploy/DeployFunctionCoreToolsStep.ts

```diff
@@ -0,0 +1,98 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type InnerDeployContext } from "@microsoft/vscode-azext-azureappservice";
+import { ActivityChildItem, ActivityChildType, activityFailContext, activityFailIcon, activityProgressContext, activityProgressIcon, activitySuccessContext, activitySuccessIcon, AzureWizardExecuteStep, createContextValue, randomUtils, type ExecuteActivityOutput } from "@microsoft/vscode-azext-utils";
+import { l10n, ThemeIcon, TreeItemCollapsibleState, type Progress } from "vscode";
+import { ext } from "../../extensionVariables";
+import { cpUtils } from "../../utils/cpUtils";
+
+export class DeployFunctionCoreToolsStep extends AzureWizardExecuteStep<InnerDeployContext> {
+    stepName: string;
+    private _childId: string = randomUtils.getRandomHexString(8); // create child id in class to make it idempotent
```

Comment:
honestly good memory that you remembered to do that lol

Reply:
This fact remained idempotent in my memory